src/pages/payments/refund/Refund.tsx

```diff
@@ -0,0 +1,213 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @copyright Copyright (c) 2021. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @copyright Copyright (c) 2021. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { Button, InputField, SelectField } from '@invoiceninja/forms';
+import axios, { AxiosError } from 'axios';
+import { endpoint } from 'common/helpers';
+import { Invoice } from 'common/interfaces/invoice';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { defaultHeaders } from 'common/queries/common/headers';
+import { usePaymentQuery } from 'common/queries/payments';
+import { Alert } from 'components/Alert';
+import Toggle from 'components/forms/Toggle';
+import { useFormik } from 'formik';
+import { useEffect, useState } from 'react';
+import toast from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from 'react-query';
+import { generatePath, useParams } from 'react-router-dom';
+
+export function Refund() {
+  const queryClient = useQueryClient();
+  const { id } = useParams();
+  const [t] = useTranslation();
+  const { data: payment } = usePaymentQuery({ id });
+  const [errors, setErrors] = useState<ValidationBag>();
+  const [invoices, setInvoices] = useState<string[]>([]);
+  const [email, setEmail] = useState(false);
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues: {
+      id: payment?.data.data.id,
+      date: payment?.data.data.date,
+
+      invoices: [],
+    },
+    onSubmit: (values) => {
+      const toastId = toast.loading(t('processing'));
+      setErrors(undefined);
+      axios
+        .post(
+          endpoint('/api/v1/payments/refund?&email_receipt=:email', { email }),
+          values,
+          {
+            headers: defaultHeaders,
+          }
+        )
+        .then(() => {
+          toast.success(t('updated_payment'), { id: toastId });
+        })
+        .catch((error: AxiosError) => {
+          console.error(error);
+          toast.error(t('error_title'), { id: toastId });
+          if (error.response?.status === 422) {
+            setErrors(error.response.data);
+          }
+        })
+        .finally(() => {
+          formik.setSubmitting(false);
+          queryClient.invalidateQueries(
+            generatePath(`/api/v1/payments/refund?&email_receipt=${email}`)
```

Comment:
`generatePath` method accepts object keys as second argument.

```js
generatePath(`/api/v1/payments/refund?&email_receipt=:email`, {email})
```

Also, change ?& to ?